packages/eslint-plugin/src/rules/consistent-generic-constructors.ts

```diff
@@ -94,7 +94,10 @@ export default createRule<Options, MessageIds>({
                   }
                   // If the property's computed, we have to attach the
                   // annotation after the square bracket, not the enclosed expression
-                  return context.sourceCode.getTokenAfter(node.key)!;
+                  return nullThrows(
+                    context.sourceCode.getTokenAfter(node.key),
+                    NullThrowsReasons.MissingToken(']', 'key'),
+                  );
```

Comment:
this pattern is so common - i wonder if we should create a utility to wrap it up and make it even shorter?

eg `getRequiredTokenAfter(sourceCode, node.key, 'optional message')`

Reply:
I like it!

Reply:
Hmm, tried this out - but in order to get the same precision of logs we'd need to still pass in `']'` and `key` as well.

```ts
return getRequiredTokenAfter(
  context.sourceCode,
  node.key,
  ']',
  'key',
);
```

I feel like I'd rather get this merged in now, then look at more options in a followup. WDYT?

Reply:
I wonder if we should just say "fuck it" and create an API without the message.
It makes the end user experience worse cos more vague error messages - but it would be much cleaner and better devx.

Reply:
I don't really mind having the extra verbosity for our devx, personally. It's been forcing me to think more carefully about what we're actually asserting on.